src/utils/service_builder.rs

```diff
@@ -207,11 +209,13 @@ fn build_backend_handlers(
 fn build_providers(
     configs: &[ProviderConfig],
     kim_factorys: HashMap<String, KeyInfoManagerFactory>,
-) -> Result<Vec<(ProviderId, Provider)>> {
+) -> Result<Vec<(ProviderId, Provider, String)>> {
     let mut list = Vec::new();
+    let mut provider_names = Vec::new();
     for config in configs {
+        // Check for duplicate providers.
         let provider_id = config.provider_id();
-        if list.iter().any(|(id, _)| *id == provider_id) {
+        if list.iter().any(|(id, _, _)| *id == provider_id) {
```

Comment:
I guess this check will be updated in a future PR to check that we have unique (provider_name, provider_uuid) tuples?

Reply:
Yes, for now I've just kept the check as is until multiple providers are properly implemented.

Reply:
Fully removed this check in the end, now just the unique `provider_name` check exists.